XbSymbolDatabase.c

```diff
@@ -84,59 +89,97 @@ static inline uint32_t BitScanReverse(uint32_t value)
 
 typedef uint8_t* memptr_t;
 
-typedef bool (*custom_scan_func_t)(uint32_t LibraryFlag,
-                                   const xbe_section_header* pSectionHeader,
-                                   unsigned short BuildVersion,
-                                   const char* LibraryStr,
-                                   xb_symbol_register_t register_func,
-                                   memptr_t xb_start_virt_addr);
+typedef struct _OutputHandler {
+    xb_output_message   verbose_level;
+    xb_output_message_t func;
+} OutputHandler;
+
+// Library Type is a requirement to prevent another thread
+// doing the same scan process.
+typedef enum _eLibraryType {
+    LT_D3D = 0,
+    LT_AUDIO,
+    LT_JVS,
+    LT_XAPI,
+    LT_GRAPHIC,
+    LT_NETWORK,
+    LT_COUNT,
+    LT_UNKNOWN=LT_COUNT
```

Comment:
It's cleaner to swap these two, so list LT_UNKNOWN first, and declare LT_COUNT as an assignment of LT_UNKNOWN.